### PR TITLE
Feature new sdk update pull of events

### LINKED ISSUE #83: No se pueden crear varias surveys para un mismo orgunit en estado "sin terminar" (añadir validación). Cuando un survey se termina, se pasa a un estado "finished" y se guarda en la DB para las Analytics

### PR DESCRIPTION
**Pull request by idelcano**
Closes #83
- Remove DataValeFlow from sdk and in malariapp side. (Instead of TrackedEntityDataValueFlow).

-Refactor pull of evens by orgunit and program using a wrapper class.

### CODE REVIEW

**Comment by ifoche** [comment]
@xurxodev. Your first PR with conflicts to resolv. Remember to merge feature-new_sdk into feature-new_sdk_update_pull_of_events (resolving conflicts) and push the resolution to feature-new_sdk_update_pull_of_events  before start doing any analysis about how this PR performs

File: app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java [hunk 1]
```diff
@@ -261,7 +261,9 @@ private static void pullOrganisationUnits() {
                     public void call(List<OrganisationUnit> organisationUnits) {
                         Log.d(TAG, "OrganisationUnit: Done");
                         asyncDownloads--;
-                        if (pullData) {
+                        mProgramAndOrganisationUnitWrapper = new ProgramAndOrganisationUnitWrapper();
+                        mProgramAndOrganisationUnitWrapper.buildProgramAndOrganisationUnit(organisationUnits, sdkPrograms);
+                        if(pullData) {
                             loadDataValues();
                         }
                         convertData();
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java]: @idelcano I have a doubt. Does it make sense to invoke ConvertData from here when is invoked from pullEventsByProgramAndOrganisationUnit?

File: app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java [hunk 2]
```diff
@@ -261,7 +261,9 @@ private static void pullOrganisationUnits() {
                     public void call(List<OrganisationUnit> organisationUnits) {
                         Log.d(TAG, "OrganisationUnit: Done");
                         asyncDownloads--;
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java]: @idelcano I think this variable management is not necessary if we invoke convertData when finish last request to api to pull events and if ConvertData is not invoked from any place more

File: app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java (hunk 1 above)
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java]: No... I removed it.

File: app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java (hunk 2 above)
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java]: Yes, it is not necessary at this point, i removed it, thanks @xurxodev

**Comment by idelcano** [comment]
@xurxodev thanks for your review, I revised your comments and updated the branch.

**Comment by xurxodev** [comment]
@idelcano Thank you for making the changes that I request you

**Comment by idelcano** [comment]
@xurxodev  It is ready, i used other rx call ( I didn't it work for my with your call, but maybe i didn't do it in the correct way)

**Comment by xurxodev** [comment]
@idelcano An timeout error occurs during the pull.

**Comment by idelcano** [comment]
@xurxodev you need try this branch with the updated events SDK branch.

**Comment by idelcano** [comment]
@xurxodev  in the sdk folder move and pull the branch : https://github.com/EyeSeeTea/dhis2-android-sdk/tree/2.25-EyeSeeTea_add_pull_of_events_by_program_and_organisationUnit

**Comment by idelcano** [comment]
@xurxodev  can you review it again? thanks

**Comment by xurxodev** [comment]
@idelcano Thanks for the effort. 
There are some things to refactor.
1 - PullUseCase must be invoked from ProgressActivity and the only thing it does is invoke pullController Through its abstraction IPullController and translate results in callbacks to ProgressActivity as in LoginUseCase and LogoutUseCase
2 - In this case the abstraction of the adapter of the data layer is not a repository must be IPullController and its implementation PullController. The goal is to take PullController to the data layer.
3- The callbacks returned by PullUseCase should not be the same as the login, it does not make sense to return, for example, InvalidCredentials because it is supposed to have already been logged in. The callbacks should be the steps "Downloading programs and orgunits ..." etc ...

The diagram we saw the other day can help you https://docs.google.com/presentation/d/1w6pc-F2cbcR1RHCE5n7IXIqomLWF0qr3lXRwmhTb7Do/edit?usp=sharing

**Comment by idelcano** [comment]
@xurxodev thanks, it is ready again!.

**Comment by xurxodev** [comment]
@idelcano I forgot one more thing to change. IPullDataSource have only one implementation then is not necessary this abstraction

**Comment by idelcano** [comment]
@xurxodev thanks for your review, I revised your comments and updated the branch

I created a ConversionLocalDataSource class, to separe the dhis db and the app db operations.

**Pull request merged**